nisystemlink/cross-service/GenericApi.py

```diff
@@ -0,0 +1,1040 @@
+        if resp.failed:
+            raise InternalError()
+
+    async def set_channel_meta(self, record_id: str, channel: str, channel_meta: Dict[str, str]):
+        """Sets the metadata for the specified channel.
+
+        :param channel: The original name of the channel
+        :param channelMeta: The dictionary representing the channel's metadata.
+        """
+        column_exist = False
+        frames = await self.get_test_meta(record_id, ["data_table_ids"])
+        for frame_id in frames["data_table_ids"]:
+            http_conn = http.client.HTTPSConnection(self._sl_uri + ":443")
+            http_conn.request("GET", "/nidataframe/v1/tables/" + frame_id, headers=self._headers)
+            resp = http_conn.getresponse()
+            response = json.loads(resp.read())
+            for column in response["columns"]:
+                if column["name"] == channel:
+                    column_exist = True
+                    body = json.dumps({"columns": [{"name": channel, "properties": channel_meta}]})
+                    http_conn.request(
+                        "PATCH",
+                        "/nidataframe/v1/tables/" + frame_id,
+                        body=body,
+                        headers=self._headers,
+                    )
+        if column_exist is False:
+            print("Channel " + channel + " doesn't exist")
+
+    def set_channel_mapping(self, channel_mappings: Dict[str, str]):
+        """Set mappings of original names to standardized names.
+
+        Sets the channel mappings of a given file. This should map the
+        file-centric names to the more commonly used aliases which
+        are referenced elsewhere.
+
+        :param channelMappings: A dictionary mapping aliases to channel names,
+        where the key represents the alias and the value represents the native
+        channel name.
+        """
+        # todo
+        raise NotImplementedError()
+        return ""
+
+    async def get_target_frame_by_name(
+        self, record_id: str, frame_names: List[str]
+    ) -> Dict[str, str]:
+        """Finds the frame_id of a given frame by name for a specific record id."""
+        target_to_source_mapping = {}
+        for source_frame in frame_names:
+            test_meta = await self.get_test_meta(record_id, ["data_table_ids"])
+            if test_meta["data_table_ids"][0] == "Not Found":
+                raise InternalError("Record has no dataframe attached")
+            for target_frame in test_meta["data_table_ids"]:
+                specific_frame_name = self._get_datatable_name(target_frame)
+                if source_frame == specific_frame_name:
+                    target_to_source_mapping[target_frame] = source_frame
+            if source_frame not in target_to_source_mapping.values():
+                raise KeyError("No corresponding target frame.")
+        return target_to_source_mapping
+
+    async def append_data(
```

Comment:
`append_data` writes data to multiple tables on a test result from a Pandas df. Interacting with the DFS using Pandas is probably the most useful thing we could add to this package. I could see us having two separate classes for reading and writing, something like `RowDataReader` and `RowDataWriter`. Usage would be something like:

```py
with RowDataWriter(chunk_size=5000, end_of_data_on_exit=True) as writer:
  # creates tables matching Pandas columns and appends data in batches
  writer.write_df('table1', df1)
  writer.write_df('table2', df2)
```